### PR TITLE
Trim unneeded fields in informer cache

### LINKED ISSUE #2279: [glbc] Only watch pods hostNetwork=false or having container ports specified
I am tempted to add a watch filter for pod informer to only watch pods with `hostNetwork=false` or having container ports specified to reduce cache footprint. However, I am not entirely sure if `glbc` is actually interested in those pods.

### PR DESCRIPTION
**Pull request by linxiulei**
Trim unneeded fields for memory efficiency. Lint check is added to prevent using those fields in the future changes.

Fixes #2279

### CODE REVIEW

**Comment by linxiulei** [comment]
`pull-ingress-gce-test` appears not respecting changes in golang-ci

File: .golangci.yaml [hunk 1]
```diff
@@ -6,18 +6,86 @@ linters:
...
+      path: pkg/context/context.go$
+    - linters:
+      - forbidigo
+      path: pkg/e2e/fixtures.go$
 linters-settings:
   errcheck:
     exclude-functions:
       - (*flag.FlagSet).Parse
       - k8s.io/apimachinery/pkg/util/wait.PollUntil
+  forbidigo:
```
Comment (wojtek-t) [inline .golangci.yaml:32]: This is nice - I didn't know such thing exists, but it definitely makes these changes much safer for the future.

**Comment by linxiulei** [comment]
Gentle ping @swetharepakula , is it unfrozen now?

**Review by swetharepakula — COMMENTED** [review]
/hold We are holding due to the code freeze

/lgtm
/approve

**Review by swetharepakula — COMMENTED** [review]
/lgtm

**Pull request merged**